notes/zrq/20210212-01-speed-tests.txt

```diff
@@ -0,0 +1,881 @@
+
+        executor-memory
+            63/3 = 21
+            21 * (1 - 0.07) = 19
+
+
+    alais
+        4 nodes
+        13 cores per node
+        44G per node
+        3 executors per node
+        executor-cores 13 / 3 = 4
+        num-executors (4*3)-1 = 11
+
+        executor-memory
+            44/3 = 14
+            14 * (1 - 0.07) = 13
+
+        ---- ----
+
+        yarn.nodemanager.resource.memory-mb     45056
+        yarn.nodemanager.resource.cpu-vcores    13
+
+        executor-cores   4
+        num-executors   11
+        executor-memory 13
+
+
+        spark-master
+        spark-defaults.conf
+
+            spark.driver.memory             13g
+            spark.yarn.am.memory            13g
+            spark.yarn.am.cores               4
+
+            spark.executor.memory           13g
+            spark.executor.cores              4
+            spark.executor.instances         11
+
+            spark.eventLog.enabled  true
+            spark.driver.maxResultSize	     8g
+
+
+        yarn-masters
+        yarn-site.xml
+            yarn.scheduler.maximum-allocation-mb 13312
+            yarn.scheduler.minimum-allocation-mb  2048
+                                                 14336
+
+
+        yarn-workers
+        yarn-site.xml
+
+            yarn.nodemanager.resource.memory-mb         ((45-1)*1024) = 45056
+            yarn.nodemanager.resource.cpu-vcores        13
+            yarn.scheduler.maximum-allocation-vcores    26
+            yarn.scheduler.minimum-allocation-vcores     1
+
+# -----------------------------------------------------
+# -----------------------------------------------------
+
+
+    https://github.com/hortonworks/hdp-configuration-utils
+
+    hdp-configuration-utils.py -c 14 -m 45 -d 1 -k False
+
+    >   Using cores=14 memory=45GB disks=1 hbase=False
+    >   Profile: cores=14 memory=45056MB reserved=1GB usableMem=44GB disks=1
+    >   Num Container=3
+    >   Container Ram=14336MB
+    >   Used Ram=42GB
+    >   Unused Ram=1GB
+    >   ***** mapred-site.xml *****
+    >   mapreduce.map.memory.mb=14336
+    >   mapreduce.map.java.opts=-Xmx11264m
+    >   mapreduce.reduce.memory.mb=14336
+    >   mapreduce.reduce.java.opts=-Xmx11264m
+    >   mapreduce.task.io.sort.mb=1792
+    >   ***** yarn-site.xml *****
+    >   yarn.scheduler.minimum-allocation-mb=14336
+    >   yarn.scheduler.maximum-allocation-mb=43008
+    >   yarn.nodemanager.resource.memory-mb=43008
+    >   yarn.app.mapreduce.am.resource.mb=14336
+    >   yarn.app.mapreduce.am.command-opts=-Xmx11264m
+    >   ***** tez-site.xml *****
+    >   tez.am.resource.memory.mb=14336
+    >   tez.am.java.opts=-Xmx11264m
+    >   ***** hive-site.xml *****
+    >   hive.tez.container.size=14336
+    >   hive.tez.java.opts=-Xmx11264m
+    >   hive.auto.convert.join.noconditionaltask.size=3758096000
+
+
+# -----------------------------------------------------
+# -----------------------------------------------------
+
+dev deployment #1
```

Comment:
Can we add some info describing the dev settings here? i.e. Spark/Hadoop settings & info on VMs

Reply:
All of the settings are in the Ansible scripts.
- `live deployment` is the existing live deployment as it was at the time
- `dev deployment` is a new deployment using the updated Ansible scripts